app/controllers/communicarts_controller.rb

```diff
@@ -21,9 +21,16 @@ def send_cart
   def approval_response
     cart = Cart.find(params[:cart_id]).decorate
     client_data = cart.proposal.client_data_legacy
-    approval = cart.approvals.find_by(user_id: user_id)
+    approval = cart.approvals.approvable.find_by(user_id: user_id)
+    
+    if !approval
+      flash[:error] = "Sorry. You are not allowed to approve your own request."
```

Comment:
Trying to think if there are other cases where the `approval` could be `nil`...if we want to show this message specifically, might be better to go back to the way you were handling it before. Sorry!